src/network/networkd-ndisc.c

```diff
@@ -207,7 +209,20 @@ static void ndisc_router_process_autonomous_prefix(Link *link, sd_ndisc_router *
         address->prefixlen = prefixlen;
         address->flags = IFA_F_NOPREFIXROUTE|IFA_F_MANAGETEMPADDR;
         address->cinfo.ifa_prefered = lifetime_preferred;
-        address->cinfo.ifa_valid = lifetime_valid;
+
+        /* see RFC4862 section 5.5.3.e */
+        r = address_get(link, address->family, &address->in_addr, address->prefixlen, &existing_address);
+        if(r > 0){
+            if(lifetime_valid > NDISC_PREFIX_LFT_MIN || lifetime_valid > existing_address->cinfo.ifa_valid)
+                address->cinfo.ifa_valid = lifetime_valid;
+            else if(existing_address->cinfo.ifa_valid <= NDISC_PREFIX_LFT_MIN)
+                address->cinfo.ifa_valid = existing_address->cinfo.ifa_valid;
+            else
+                address->cinfo.ifa_valid = NDISC_PREFIX_LFT_MIN;
+        }else if(lifetime_valid > 0)
+            address->cinfo.ifa_valid = lifetime_valid;
```

Comment:
lifetime_valid doesn´t need to be 2 hours or longer. That means the router can add a prefix/address with a lifetime of 30 secs and renew it every 10 secs. This behavior is totally fine but a bogus RA can´t force it to invalidate faster. If we got new prefix (address_get() < 0) we can´t decide if this is a legit Prefix or not, so the only option is to assume it is, and if it is the route updates the prefix before it gets invalid.